### DISCUSSION TITLE
[Pedido de Funcionalidade] Adicionar Memória de Pesquisa na Caixa de Busca

### DISCUSSION THREAD

**Discussion by esmsena** [vercel/next.js#68557] [Ideas]
### Link to the code that reproduces this issue

https://github.com/filipedeschamps/tabnews.com.br/blob/main/pages/interface/components/SearchBox/index.js

#### Passos para Reproduzir
1. Acesse a aplicação.
2. Utilize a caixa de busca para pesquisar qualquer termo.
3. Feche a aplicação ou vá para outra página.
4. Retorne à página com a caixa de busca e tente pesquisar novamente.
5. Verá que o que pesquisou não aparece como busca recente na caixa de busca

#### Comportamento Atual:
As pesquisas anteriores não são exibidas como sugestões ou histórico na caixa de busca.

#### Comportamento Esperado:
As pesquisas anteriores devem ser armazenadas e exibidas como sugestões ou histórico quando o usuário acessa a caixa de busca novamente.

#### Primeira imagem mostrando pesquisa feita 

![Captura de ecrã de 2024-07-30 20-44-38](https://github.com/user-attachments/assets/c29554f4-4799-4d0a-a16a-e895c4be6fae)

#### Segunda imagem após pesquisa "jogo" , a caixa de pesquisa não mostra nada pesquisado anteriormente 

![image](https://github.com/user-attachments/assets/e7f8a53d-b2ae-4461-a898-fe53bdeff4b8)

Implementar uma funcionalidade que armazene as pesquisas realizadas pelos usuários em um armazenamento local (por exemplo, localStorage no navegador) e exibi-las como sugestões ou histórico na próxima vez que o usuário acessar a caixa de busca.

### Provide environment information

```bash
Navegador: Chrome 91.0.4472.124
Sistema Operacional: Ubuntu 22.04.4 LTS
Node.js: v22.3.0
NPM: 10.8.1
Relevant Packages:
  next: 15.0.0-canary.91 // Latest available version is detected (15.0.0-canary.91).
  eslint-config-next: N/A
  react: 19.0.0-rc.0
  react-dom: 19.0.0-rc.0
  typescript: 5.3.3
```

### Which area(s) are affected? (Select all that apply)

Navigation

### Which stage(s) are affected? (Select all that apply)

Other (Deployed)

**Comment by samcx** [comment]
Hi @esmsena!

A "feature request" should be added to our [Discussions](https://github.com/vercel/next.js/discussions). Converting this to a discussion.
